shell/platform/android/io/flutter/embedding/android/AndroidKeyProcessor.java

```diff
@@ -94,4 +161,103 @@ private Character applyCombiningCharacterToBaseCharacter(int newCharacterCodePoi
 
     return complexCharacter;
   }
+
+  public static class EventResponder implements KeyEventChannel.EventResponseHandler {
+    // The maximum number of pending events that are held before starting to
+    // complain.
+    private static final long MAX_PENDING_EVENTS = 1000;
+    final Deque<Entry<Long, KeyEvent>> pendingEvents = new ArrayDeque<Entry<Long, KeyEvent>>();
+    @NonNull private final Context context;
+    boolean dispatchingKeyEvent = false;
```

Comment:
Can we make all non-public properties private, and only test public features, so that the implementation can be changed in the future without breaking anything?

Reply:
I think it'd be better to only test how this class interact with other classes.

In this case, we would like a `AndroidKeyProcessor` to always return true for `onKeyDown/onKeyUp`, which sends the event through a method channel. If the method channel call reports unhandled, we expect the event to be received by `AndroidKeyProcessor` again, but this time it returns false.

Everything else `AndroidKeyProcessor` uses for this process is implementation details, including probably the entire `EventResponder` class and its properties. We can write tests for them all we want, but these tests probably mean little, because if some other demands come and this implementation needs changing, such as changing the boolean into an integer, this change wouldn't even break anything outside of this file, but many tests for this file become invalid.

Instead of testing the private boolean flag itself, I think it's better to test what it means. For example, `dispatchingKeyEvent` means that during an `EventResponder. dispatchKeyEvent` the `AndroidKeyProcessor.onKeyUp/onKeyDown` will return false. If tests can ensure this is right, it really doesn't matter if the boolean is renamed or changed to its negated version or removed at all.

Reply:
Sure, I get it. "Test behaviors, not implementations".

I was trying to get at the exact behavioral indicator to make sure that the mechanism is working, which is internal, but your point is valid. Without looking at internal information, it's not possible to write a test that proves that it will always return false during the handling of an event, but I suppose that's not required.

Reply:
OK, I've made all of the implementation of AndroidKeyProcessor private, and I'm just testing the public effects.